ckanext/unaids/tests/test_actions.py

```diff
@@ -192,3 +191,22 @@ def test_user(self):
         user_obj = model.User.get(user['name'])
         response = call_action('user_show_me', {'auth_user_obj': user_obj})
         assert response['name'] == user['name']
+
+
+@pytest.mark.ckan_config('ckan.plugins', 'unaids')
+@pytest.mark.usefixtures('with_plugins')
+class TestPopulateDataDictionary(object):
+
+    def test_expected_use(self, mocker):
+        dataset = factories.Dataset()
+        resource = factories.Resource(
+            package_id=dataset['id'],
+            schema='test_schema',
+            url='some.data'
```

Comment:
@tomeksabala I have had to add this line in because my test broke after rebasing against development for no reason that I could understand.  Some further inspection suggests perhaps it is a bug in your filename munging solution here: https://github.com/fjelltopp/ckanext-unaids/blob/a0c6b9af5d5afd0e6551600ca114cb4be12282c9/ckanext/unaids/logic.py#L35

Basically the default url value supplied by the factory is `http://link.some.data` I think your code keeps adding extra copies of `http://` to the start of the strin i.e `http://http://http://link.some.value` depending how often it is hit.  Can you investigate?  To reproduce just remove this line and re run this test.

Reply:
ok, I'll have a look into that.

Reply:
I have a fix for that but let's merge it in a sep PR after this one.